tests/PHPStan/Rules/Comparison/data/boolean-xor.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace BooleanXor;
+
+function leftAlwaysTrue(bool $right): bool
+{
+	return true xor $right;
+}
+
+function rightAlwaysTrue(bool $left): bool
+{
+	return $left xor true;
+}
+
+function leftAlwaysFalse(bool $right): bool
+{
+	return false xor $right;
+}
+
+function rightAlwaysFalse(bool $left): bool
+{
+	return $left xor false;
+}
+
+/**
+ * @param true $left
+ * @param true $right
+ */
+function phpdocAlways(bool $left, bool $right): bool
+{
+	return $left xor $right;
+}
+
+function skip(\DateTimeImmutable $date): void
```

Comment:
i could not figure out a way to write an `xor` statement that had neither the left or right being always true or false, while the full statement itself is a boolean constant result.

i tried this but due to the way php models it's types as possibilities and not as branches it doesn't work :( 

```
$val = (bool) rand(0, 1);
$notVal = !$val;

$res = ($val xor $notVal);
```

if you have any wisdom let me know :)